### PR TITLE
Stop useless minimap updates and thread spam

### PR DESCRIPTION
**Pull request by mir3x**
Looks like working well

### CODE REVIEW

File: client/gui-qt/minimap.cpp [hunk 1]
```diff
@@ -272,7 +272,8 @@ void minimap_view::update_image()
   if (isHidden()) {
     return;
   }
-  thread.render(scale_factor, width(), height());
+  QTimer::singleShot(
+      10, [this] { thread.render(scale_factor, width(), height()); });
```
Comment (lmoureaux) [inline client/gui-qt/minimap.cpp:276]: What if `this` is deleted before the timer fires?

Comment (mir3x) [reply client/gui-qt/minimap.cpp]: No one will ever know :  it doesnt exist anymore

Comment (lmoureaux) [reply client/gui-qt/minimap.cpp]: `thread`, `width` or `height` may crash since they're not pointing anywhere…

Comment (mir3x) [reply client/gui-qt/minimap.cpp]: what u mean? There is no lambda anymore.

Comment (lmoureaux) [reply client/gui-qt/minimap.cpp]: The lambda stays there (in the Qt event loop mechanism) and its captured `this` becomes a dangling pointer if the object it points to gets deleted. [Apparently](https://stackoverflow.com/a/19721153) one needs to use `connect` and set the "context" of the connection, or wrap it into a `QPointer` and check it's still alive in the lambda. Bit annoying but necessary.

Comment (mir3x) [reply client/gui-qt/minimap.cpp]: :-1: 
Read what I wrote, there is no lambda. Someone deleted lambda! Poor lambda.

Comment (lmoureaux) [reply client/gui-qt/minimap.cpp]: lol sorry :P poor lambda indeed, rip

**Pull request merged**